integration-tests/manifests/annotations/validate_annotations_namespace_test.go

```diff
@@ -127,3 +132,300 @@ func TestPythonOnlyNamespace(t *testing.T) {
+			t.Fatalf("Failed to delete namespaces/resources: %v", err)
+		}
+	}()
+
+	annotationConfig := auto.AnnotationConfig{
+		Java: auto.AnnotationResources{
+			Namespaces:   []string{uniqueNamespace},
+			DaemonSets:   []string{""},
+			Deployments:  []string{""},
+			StatefulSets: []string{""},
+		},
+		Python: auto.AnnotationResources{
+			Namespaces:   []string{""},
+			DaemonSets:   []string{""},
+			Deployments:  []string{""},
+			StatefulSets: []string{""},
+		},
+	}
+	jsonStr, err := json.Marshal(annotationConfig)
+	if err != nil {
+		t.Error("Error:", err)
+	}
+
+	startTime := time.Now()
+	updateTheOperator(t, clientSet, string(jsonStr))
+	if err != nil {
+		t.Errorf("Failed to get deployment app: %s", err.Error())
+	}
+	deployment, err := clientSet.AppsV1().Deployments(uniqueNamespace).Get(context.TODO(), deploymentName, metav1.GetOptions{})
+	if err != nil {
+		fmt.Printf("Error retrieving deployment: %v\n", err)
+		os.Exit(1)
+	}
+	nginxDeployment, err := clientSet.AppsV1().Deployments(uniqueNamespace).Get(context.TODO(), nginxDeploymentName, metav1.GetOptions{})
+	if err != nil {
+		fmt.Printf("Error retrieving deployment: %v\n", err)
+		os.Exit(1)
+	}
+
+	err = util.WaitForNewPodCreation(clientSet, deployment, startTime)
+	if err != nil {
+		fmt.Printf("Error waiting for pod creation: %v\n", err)
+		os.Exit(1)
+	}
+
+	if annotationExists(nginxDeployment.Annotations, autoAnnotateJavaAnnotation) {
+		t.Fatal("Auto-annotation annotation should not exist")
+	}
+
+	numOfRevisions, err := numberOfRevisions(nginxDeploymentName, uniqueNamespace)
+	if numOfRevisions > 1 || err != nil {
+		t.Fatal("Nginx was restarted") //should not be restarted since it already had annotations
+	}
+	numOfRevisions, err = numberOfRevisions(deploymentName, uniqueNamespace)
+	if numOfRevisions != 2 || err != nil {
+		t.Fatal("Sample-deployment should have been restarted") //should not be restarted since it already had annotations
+	}
+
+}
+
+// Test if a resource is auto annotated and now its namespace is added for auto annotation
+// the resource should not be restarted
+func TestAlreadyAutoAnnotatedResourceShouldNotRestart(t *testing.T) {
+
+	clientSet := setupTest(t)
+	randomNumber, err := rand.Int(rand.Reader, big.NewInt(9000))
+	if err != nil {
+		panic(err)
+	}
+	randomNumber.Add(randomNumber, big.NewInt(1000)) //adding a hash to namespace
+	uniqueNamespace := fmt.Sprintf("multiple-resources-%d", randomNumber)
+	startTime := time.Now()
+	if err := createNamespaceAndApplyResources(t, clientSet, uniqueNamespace, []string{sampleDeploymentYamlNameRelPath}); err != nil {
+		t.Fatalf("Failed to create/apply resoures on namespace: %v", err)
+	}
+	defer func() {
+		if err := deleteNamespaceAndResources(clientSet, uniqueNamespace, []string{sampleDeploymentYamlNameRelPath}); err != nil {
+			t.Fatalf("Failed to delete namespaces/resources: %v", err)
+		}
+	}()
```

Comment:
This seems like a common block of setup code. Consider moving it into a helper function.

Reply:
makes sense, I put everything above the defer func in the helper. I changed the defer func to t.cleanup and I have to make sure it is in the original test function otherwise the deployment will be deleted, but the rest is in helper func.